src/main/java/com/cairnfg/waypoint/authorization/endpoints/account/RemoveCoClientEndPoint.java

```diff
@@ -1,72 +1,134 @@
-     * @return A ResponseEntity containing the result of the operation.
-     */
-    @Transactional
-    @PatchMapping(PATH)
-    @PreAuthorize("hasAnyAuthority('SCOPE_account.full', 'SCOPE_admin.full')")
-    @Operation(
-            summary = "Removes Co-client from household",
-            description = "Allows a user to remove a co-client from a household. Requires the `account.create` permission.",
-            security = @SecurityRequirement(name = "oAuth2JwtBearer"),
-            responses = {
-                    @ApiResponse(responseCode = "200", description = "OK - Account update was successful"),
-                    @ApiResponse(responseCode = "404", description = "Not Found",
-                            content = {@Content(mediaType = "application/json",
-                                    schema = @Schema(implementation = ErrorMessage.class))}),
-                    @ApiResponse(responseCode = "409", description = "Conflict",
-                            content = {@Content(mediaType = "application/json",
-                                    schema = @Schema(implementation = ErrorMessage.class))})
-            }
-    )
-    public ResponseEntity<?> removeCoClient(@PathVariable Long accountId, Principal principal) {
-        log.info("User [{}] is attempting to remove co-client for account [{}]", principal.getName(), accountId);
-        try {
-            accountService.removeCoClientFromHousehold(accountId);
-            return ResponseEntity.status(HttpStatus.OK)
-                    .body("Account with ID [" + accountId + "] successfully removed from household");
-        } catch (IllegalStateException e) {
-            log.warn(e.getMessage());
-            return ResponseEntity.status(HttpStatus.CONFLICT).body(e.getMessage());
-        } catch (IllegalArgumentException e) {
-            log.warn(e.getMessage());
-            return ResponseEntity.status(HttpStatus.NOT_FOUND).body(e.getMessage());
-        }
+    Account accountToUpdate = accountToUpdateOptional.get();
+
+    // Check if isPrimaryContact is false, and if so, set householdId to null
+    if (Boolean.FALSE.equals(accountToUpdate.getIsPrimaryContactForHousehold())) {
+      accountToUpdate.setHousehold(null);
+      log.info("householdId set to null", accountId);
+    } else {
+      return generateFailureResponse("Account [" + accountId + "] is the primary contact",
+          HttpStatus.NOT_FOUND);
     }
+
+    Account updatedAccount = this.accountService.saveAccount(accountToUpdate);
+
+    log.info("Account [{}] updated successfully with ID [{}]", updatedAccount.getUsername(),
+        updatedAccount.getId());
+    return ResponseEntity.status(HttpStatus.OK)
+        .body("Account with ID [" + updatedAccount.getId() + "] updated successfully");
+  }
+
+  private ResponseEntity<ErrorMessage> generateFailureResponse(String message, HttpStatus status) {
+    log.warn(message);
+    TransactionAspectSupport.currentTransactionStatus().setRollbackOnly();
```

Comment:
This transaction rollback can be buggy. It does not always behave how you would expect